test/extensions/filters/network/common/redis/client_impl_test.cc

```diff
@@ -32,6 +35,13 @@ namespace Common {
 namespace Redis {
 namespace Client {
 
+class MockDispatcherWithTimeOverride : public Event::MockDispatcher {
```

Comment:
I believe that you can just derive your test from `TestUsingSimulatedTime` (search for other examples) and avoid doing this. Simulated time will automatically be injected into your test.

Reply:
Excellent, I had not seen that used before- that makes this sort of test much easier!